src/SFA.DAS.FAT.Web/Models/CoursesProviderViewModel.cs

```diff
@@ -72,7 +123,15 @@ public static implicit operator CoursesProviderViewModel(ProviderData source)
             EmployerRating = source.EmployerRating,
             ApprenticeReviews = source.ApprenticeReviews,
             ApprenticeStars = source.ApprenticeStars,
-            ApprenticeRating = source.ApprenticeRating
+            ApprenticeRating = source.ApprenticeRating,
+            IsEmployerLocationAvailable = source.Locations.Any(x => x.AtEmployer),
+            NearestEmployerLocation = source.Locations.FirstOrDefault(x => x.AtEmployer)?.CourseDistance,
+            IsBlockReleaseAvailable = source.Locations.Any(x => x.BlockRelease),
+            IsDayReleaseAvailable = source.Locations.Any(l => l.DayRelease),
+            IsBlockReleaseMultiple = source.Locations.Count(x => x.BlockRelease) > 1,
+            IsDayReleaseMultiple = source.Locations.Count(x => x.DayRelease) > 1,
+            NearestBlockRelease = source.Locations.FirstOrDefault(x => x.BlockRelease)?.CourseDistance,
```

Comment:
The point was to do FirstOrDefault once and then use that item to populate the properties.

Reply:
you can't use the general FirstOrDefauilt to find the nearest day release etc because the nearest day release etc might not be in the FirstOrDefault it might be second or third etc.  Unless I've misunderstood what you're suggesting.